File: graphql-test/src/main/java/com/junyss/graphqltest/software/repository/SoftwareRepository.java
package com.junyss.graphqltest.software.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import com.junyss.graphqltest.software.model.entity.Software;

public interface SoftwareRepository extends JpaRepository<Software, String> {
	@Query(value = "select software from Software software where software.softwareId = :softwareId")
	Optional<Software> findBySoftwareId(String softwareId);
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/software/repository/SoftwareRepository.kt
package com.junyharangstudy.kotlingraphqltest.api.software.repository

import com.junyharangstudy.kotlingraphqltest.api.software.model.entity.Software
import org.springframework.data.jpa.repository.JpaRepository

interface SoftwareRepository: JpaRepository<Software, String>